shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -25,4 +28,50 @@ class ExpandedEntity private constructor(
 
     fun compact(): Map<String, Any> =
         JsonLdProcessor.compact(attributes, mapOf("@context" to contexts), JsonLdOptions())
-}
+
+    private fun getAttributeOfType(type: AttributeType) =
```

Comment:
It's better to specify the return type, to make clear what we get from the method (and call it `getAttributesOfType` because we typically have more than one).

It could also be interesting to have the properties and relationships as lazy properties (https://kotlinlang.org/docs/reference/delegated-properties.html#lazy).